src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/IntellisenseQuickInfoBuilder.cs

```diff
@@ -79,16 +107,67 @@ internal static class IntellisenseQuickInfoBuilder
             }
 
             var content = new ContainerElement(
-                                ContainerElementStyle.Stacked,
+                                ContainerElementStyle.Stacked | ContainerElementStyle.VerticalPadding,
                                 elements);
 
             return new IntellisenseQuickInfoItem(trackingSpan, content);
         }
 
-        private static ClassifiedTextElement BuildClassifiedTextElement(QuickInfoSection section)
+        private static IEnumerable<object> BuildClassifiedTextElements(QuickInfoSection section)
         {
-            return new ClassifiedTextElement(section.TaggedParts.Select(
-                    part => new ClassifiedTextRun(part.Tag.ToClassificationTypeName(), part.Text)));
+            var paragraphs = new List<object>();
+            var currentParagraph = new List<object>();
+            var currentRuns = new List<ClassifiedTextRun>();
+            foreach (var part in section.TaggedParts)
+            {
+                if (part.Tag == TextTags.LineBreak)
+                {
+                    if (currentRuns.Count == 0)
+                    {
+                        if (currentParagraph.Count > 0)
+                        {
+                            if (currentParagraph.Count == 1)
+                            {
+                                paragraphs.Add(currentParagraph[0]);
+                            }
+                            else
+                            {
+                                paragraphs.Add(new ContainerElement(ContainerElementStyle.Stacked, currentParagraph));
+                            }
+
+                            currentParagraph.Clear();
+                        }
+                    }
+                    else
+                    {
+                        currentParagraph.Add(new ClassifiedTextElement(currentRuns));
+                        currentRuns.Clear();
+                    }
+                }
+                else
+                {
+                    currentRuns.Add(new ClassifiedTextRun(part.Tag.ToClassificationTypeName(), part.Text));
```

Comment:
it seems could become generic helper extension method if we don't already have one. TaggedText.ToClassifiedText()

Reply:
any taggedtext before line breaks and accumulated in current run

Reply:
> it seems could become generic helper extension method if we don't already have one

We didn't already have one, but this code is only used by this IntelliSense builder so it didn't seem to improve clarity to move this to another file. I did end up refactoring the pattern used for moving content from `currentParagraph`&rarr;`paragraphs`.

Reply:
I pushed the refactoring and documentation update for `BuildClassifiedTextElements`, so it should be much more clear within that method.